src/frontend/src/lib/services/address.services.ts

```diff
@@ -120,21 +132,21 @@ const loadIdbTokenAddress = async <T extends Address>({
 	tokenId: TokenId;
 	getIdbAddress: (principal: Principal) => Promise<IdbAddress<T> | undefined>;
 	updateIdbAddressLastUsage: (principal: Principal) => Promise<void>;
-}): Promise<ResultSuccess> => {
+}): Promise<ResultSuccess<TokenId>> => {
 	try {
 		const { identity } = get(authStore);
 
 		// Should not happen given the current layout and guards.
 		assertNonNullish(identity, 'Cannot continue without an identity.');
 
 		if (identity.getPrincipal().isAnonymous()) {
-			return { success: false };
+			return { success: false, err: tokenId };
```

Comment:
Even though it is type as `unknown`, if you wish to use the `err` to return the token id, I would then suggest to create a class that extends `Error` and return it as it's content. This will make the usage more clear, particularly as the error is consumed and interpreted in the component, and also better typed when transpiled to JavaScript as well.

Reply:
maybe I understood what you mean: please check last commit